classes/class-wc-connect-functions.php

```diff
@@ -61,11 +61,73 @@ public static function is_store_api_call() {
 		}
 
 		/**
-		 * Check if current page has woocommerce cart or checkout block.
+		 * Check if current page is a cart page or has woocommerce cart block.
 		 *
 		 * @return bool
 		 */
-		public static function has_cart_or_checkout_block() {
+		public static function is_cart() {
+			if ( is_cart() || self::has_cart_block() ) {
+				return true;
+			}
+
+			return false;
+		}
+
+		/**
+		 * Check if current page is a checkout page or has woocommerce checkout block.
+		 *
+		 * @return bool
+		 */
+		public static function is_checkout() {
+			if ( is_checkout() || self::has_checkout_block() ) {
+				return true;
+			}
+
+			return false;
+		}
+
+		/**
+		 * Check if current page has woocommerce cart block.
+		 *
+		 * @return bool
+		 */
+		public static function has_cart_block() {
+			// To support WP < 5.0.0, we need to check if `has_block` exists first as has_block only being introduced on WP 5.0.0.
+			if ( function_exists( 'has_block' ) ) {
+				return has_block( 'woocommerce/cart' );
+			}
+
+			$page = get_post();
+			if ( ! $page ) {
+				return false;
+			}
+
+			$blocks = parse_blocks( $page->post_content );
+			if ( ! $blocks ) {
+				return false;
+			}
+
+			foreach ( $blocks as $block ) {
+				$block_name = $block['blockName'];
+				if ( 'woocommerce/cart' === $block_name ) {
+					return true;
+				}
+			}
+
+			return false;
+		}
```

Comment:
```suggestion
		public static function has_cart_block() {
			// To support WP < 5.0.0, we need to check if `has_block` exists first as has_block only being introduced on WP 5.0.0.
			if ( function_exists( 'has_block' ) ) {
				return has_block( 'woocommerce/cart' );
			}

			return false;
		}
```

Reply:
Good catch @dustinparker! thanks for catching this. I will shorten this function.